fnl/init.fnl

```diff
@@ -33,6 +44,27 @@
       t
       nil)))
 
+(fn append [lhs v]
+  (var nl [])
+  (each [k v (ipairs lhs)]
+    (tset nl k v))
+  (table.insert nl v)
+  nl)
+
+(global cons
+  (fn [_lhs _rhs]
+    (assert (not (in (type! _lhs) [:function :table])))
+    (assert (not (in (type! _rhs) [:function :table])))
+    (var lhs _lhs)
+    (var rhs _rhs)
+    (when (in (type! lhs) [:number :string])
+      (set lhs [lhs]))
+    (when (in (type! rhs) [:number :string])
+      (set rhs [rhs]))
+    (if (= 0 (length rhs))
+      lhs
+      (cons (append lhs (car rhs)) (cdr rhs)))))
```

Comment:
`append` should rely on `cons` not the other way around

Reply:
Also `(cons [1 2] [3 4])` returns `[1 2 3 4]` when it should return `[[1 2] [3 4]]`

Reply:
here https://github.com/fedemengo/nvim/issues/12